src/rpc/state_api.rs

```diff
@@ -176,6 +177,22 @@ pub(in crate::rpc) async fn state_miner_power<DB: Blockstore + Send + Sync + 'st
         .map_err(|e| e.into())
 }
 
+/// looks up the miner power of the given address.
+pub(in crate::rpc) async fn state_miner_faults<DB: Blockstore + Send + Sync + 'static>(
+    data: Data<RPCState<DB>>,
+    Params(LotusJson((address, key))): Params<LotusJson<(Address, TipsetKeys)>>,
+) -> Result<LotusJson<BitField>, JsonRpcError> {
+    let ts = data
+        .state_manager
+        .chain_store()
+        .load_required_tipset(&key)?;
+
+    data.state_manager
+        .miner_faults(&address, &ts)
+        .map_err(|e| e.into())
+        .map(|r| r.into())
```

Comment:
nit: maybe make it more explicit what types we're converting into.
```suggestion
    Ok(LotusJson(data.state_manager.miner_faults(&address, &ts)?))
```

Reply:
It lines up with the other endpoints, I see no reason to be explicit here, while being implicit in other methods.

Reply:
Also, we'll still need to do `map_err`.

Reply:
Why do we need `map_err`?

Reply:
Ah, we don't, if we use the `?`. I've basically used the same approach I did for the `StateMinerPower`, which was used in `state_market_balance`. 

I'd say that it would be nice to go with the same approach for all the methods. I've just taken a look and we're pretty much 50/50. In my opinion the best course of action here is to let everybody go with their own flow and once we've finished the RPC compatibility routine - just do a small refactoring to make them look the same.